### PR TITLE
Implement lazy part of StaticStrings

### PR DESCRIPTION
**Pull request by ksh8281**

### CODE REVIEW

File: src/runtime/GlobalObjectBuiltinIntl.cpp [hunk 1]
```diff
@@ -525,7 +522,7 @@ static Value builtinIntlPluralRulesResolvedOptions(ExecutionState& state, Value
         i++;
     } while (true);
 
-    options->defineOwnPropertyThrowsException(state, ObjectPropertyName(state, Value(String::fromASCII("pluralCategories"))), ObjectPropertyDescriptor(pluralCategories, ObjectPropertyDescriptor::AllPresent));
+    options->defineOwnPropertyThrowsException(state, ObjectPropertyName(AtomicString(state.context(), "pluralCategories")), ObjectPropertyDescriptor(pluralCategories, ObjectPropertyDescriptor::AllPresent));
```
Comment (clover2123) [inline src/runtime/GlobalObjectBuiltinIntl.cpp:525]: Why didn't you convert "pluralCategories" into lazy string?

File: src/runtime/StaticStrings.cpp [hunk 2]
```diff
@@ -136,8 +137,25 @@ void StaticStrings::initStaticStrings(AtomicStringMap* atomicStringMap)
...
+#define DECLARE_LAZY_STATIC_STRING(Name, unused) m_lazy##Name = AtomicString();
+    FOR_EACH_LAZY_STATIC_STRING(DECLARE_LAZY_STATIC_STRING);
+    FOR_EACH_LAZY_INTL_STATIC_STRING(DECLARE_LAZY_STATIC_STRING);
+#undef DECLARE_LAZY_STATIC_STRING
 }
 
+#define DECLARE_LAZY_STATIC_STRING(Name, stringContent)                                                                           \
+    AtomicString StaticStrings::lazy##Name()                                                                                      \
+    {                                                                                                                             \
+        if (m_lazy##Name.string() == String::emptyString) {                                                                       \
```
Comment (clover2123) [inline src/runtime/StaticStrings.cpp:150]: How about adding `UNLIKELY` macro at this point?

File: src/runtime/GlobalObjectBuiltinIntl.cpp (hunk 1 above)
Comment (ksh8281) [reply src/runtime/GlobalObjectBuiltinIntl.cpp:525]: "pluralCategories" is only used here.
and resolvedOptions function is called rarely I think(this function is not used on initializion or formatting).
so I didn't want use one pointer space for "pluralCategories"

File: src/runtime/StaticStrings.cpp (hunk 2 above)
Comment (ksh8281) [reply src/runtime/StaticStrings.cpp:150]: Good idea!

Comment (ksh8281) [reply src/runtime/StaticStrings.cpp]: I added it

File: src/runtime/GlobalObjectBuiltinIntl.cpp (hunk 1 above)
Comment (clover2123) [reply src/runtime/GlobalObjectBuiltinIntl.cpp:525]: I see. BTW how about using `String::FromExternalMemory` flag to avoid any string copy for this case?

Comment (ksh8281) [reply src/runtime/GlobalObjectBuiltinIntl.cpp:525]: Good idea!

Comment (ksh8281) [reply src/runtime/GlobalObjectBuiltinIntl.cpp:525]: I added it

**Pull request merged**